pkg/issuer/acme/dns/azuredns/azuredns.go

```diff
@@ -68,16 +68,31 @@ func NewDNSProviderCredentials(environment, clientID, clientSecret, subscription
 		return nil, err
 	}
 
-	spt, err := adal.NewServicePrincipalToken(*oauthConfig, clientID, clientSecret, env.ResourceManagerEndpoint)
-	if err != nil {
-		return nil, err
+	var token *adal.ServicePrincipalToken
+	if clientID != "" {
+		klog.Info("azuredns authenticating with clientID and secret key")
+		token, err = adal.NewServicePrincipalToken(*oauthConfig, clientID, clientSecret, env.ResourceManagerEndpoint)
+		if err != nil {
+			return nil, err
+		}
+	} else {
+		klog.Info("azuredns authenticating with managed identity")
+		endpoint, err := adal.GetMSIVMEndpoint()
+		if err != nil {
+			return nil, err
+		}
+
+		token, err = adal.NewServicePrincipalTokenFromMSI(endpoint, env.ServiceManagementEndpoint)
```

Comment:
I think we need to check if 'ambient credentials' are allowed to be used here too - you can pass it through from [here](https://github.com/jetstack/cert-manager/blob/cb4dbe0d94fce4dabb217b1f5c5e3706a4b9499c/pkg/issuer/acme/dns/dns.go#L193) to the provider constructor 😄